cupyx/scipy/special/digamma.py

```diff
@@ -0,0 +1,199 @@
+import cupy
+from cupy import core
+
+
+_digamma_kernel = None
+
+polevl_definition = '''
+static __device__ double polevl(double x, double coef[], int N)
+{
+    double ans;
+    int i;
+    double *p;
+
+    p = coef;
+    ans = *p++;
+    i = N;
+
+    do
+    ans = ans * x + *p++;
+    while (--i);
+
+    return (ans);
+}
+'''
+
+
+psi_definition = '''
+static __device__ double A[] = {
+    8.33333333333333333333E-2,
+    -2.10927960927960927961E-2,
+    7.57575757575757575758E-3,
+    -4.16666666666666666667E-3,
+    3.96825396825396825397E-3,
+    -8.33333333333333333333E-3,
+    8.33333333333333333333E-2
+};
+
+static __device__ double PI = 3.141592653589793;
+static __device__ double EULER = 0.5772156649015329;
+
+static __device__ double digamma_imp_1_2(double x)
+{
+    /*
+     * Rational approximation on [1, 2] taken from Boost.
+     *
+     * Now for the approximation, we use the form:
+     *
+     * digamma(x) = (x - root) * (Y + R(x-1))
+     *
+     * Where root is the location of the positive root of digamma,
+     * Y is a constant, and R is optimised for low absolute error
+     * compared to Y.
+     *
+     * Maximum Deviation Found:               1.466e-18
+     * At double precision, max error found:  2.452e-17
+     */
+    double r, g;
+
+    static const float Y = 0.99558162689208984f;
+
+    static const double root1 = 1569415565.0 / 1073741824.0;
+    static const double root2 = (381566830.0 / 1073741824.0) / 1073741824.0;
+    static const double root3 = 0.9016312093258695918615325266959189453125e-19;
+
+   static double P[] = {
+       -0.0020713321167745952,
+       -0.045251321448739056,
+       -0.28919126444774784,
+       -0.65031853770896507,
+       -0.32555031186804491,
+       0.25479851061131551
+   };
+   static double Q[] = {
+       -0.55789841321675513e-6,
+       0.0021284987017821144,
+       0.054151797245674225,
+       0.43593529692665969,
+       1.4606242909763515,
+       2.0767117023730469,
+       1.0
+   };
```

Comment:
It's great if we can inline these constant values (but I do not come up with a good way to let the compiler do that).